### DISCUSSION TITLE
Image Thumbnail Generation in background (worker)

### DISCUSSION THREAD

**Discussion by hethehe** [pimcore/pimcore#13528] [Ideas]
Wouldn't it be a good idea to generate the thumbnails in the background like the video "thumbnails" (or at least have an option to change it)?

For example, if you open a site with 100 images for the first time every thumbnail has to be generated, especially with the newer .avif format this takes a pretty long time and most likely some PHP processes will run out of memory (fails).
=> even worse, if you don't caught this, the page is no longer responsible (all php-fpm children are busy).

To handle broken images, there could be an attribute "thumbnail_available" on the thumbnail object which could be used to check if thumbnail is already generated or not.

Or am I missing something, and there is already a functionality for this?
Thank you for your feedback/ideas.

**Comment by dpfaffenbauer** [comment]
Good idea. What image should be display when it's not available yet? none? fallback? Pimcore's "error" image?

**Reply by hethehe** [reply]
I would make it configurable with a fallback image, if not set "none". Or just none because it's not the Pimcore's job to handle the frontend.

Not sure how it's done on the video "thumbnail" (generated video) but I've seen that there is already a similar solution - would be nice if they both behave the same way.
